lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1404,25 +1404,43 @@ class FlattenCat : public mlir::RewritePattern {
         worklist.push_back(operand);
     };
     pushOperands(cat);
+    bool existsSigned = false, existsUnsigned = false;
```

Comment:
Mega-nit: Using `hasSigned` and `hasUnsigned` may save enough characters that the LHS ternary assignment on line 1413 may fie on one line. 🫠